script.rb

```diff
@@ -0,0 +1,184 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    length.times do
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    my_new_arr = [] # It will return a new array
+    length.times do
+      value = yield self[i]
+      my_new_arr.push self[i] unless value == true
+      i += 1
+    end
+    my_new_arr
+  end
+
+  def my_all?
+    return to_enum(:my_all?) unless block_given?
+
+    i = 0
+    length.times do
+      value = yield self[i]
+      return false if value == false
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    i = 0
+    length.times do
+      value = yield self[i]
+      return true if value == true
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    i = 0
+    length.times do
+      value = yield self[i]
+      return false if value == true
+
+      i += 1
+    end
+    true
+  end
```

Comment:
- [ ] my_any?, my_all? and my_none? does not behave like their original enumerable methods. Again, study the original methods and make your own method behave exactly like the original methods behave.